### PR TITLE
stealth added to mixer

### PR DESCRIPTION
**Pull request by aragogi**

### CODE REVIEW

File: mixer/app/Configs.scala [hunk 1]
```diff
@@ -55,5 +55,11 @@ object Configs extends ConfigHelper {
   lazy val stopMixingWhenReachedThreshold: Boolean = readKey("stopMixingWhenReachedThreshold", "true").toBoolean
 
   lazy val hopRounds: Int = readKey("hopRounds", "0").toInt
+  lazy val bestBlockId: String = readKey("bestBlockId")
+  lazy val stealthImplementorFee: Long = readKey("fees.stealthImplementorFee").toLong
```
Comment (anon2020s) [inline mixer/app/Configs.scala]: These fees (stealthImplementorFee, stealthTransactionFeePercent) should be hard-code at this file, also It should not be included in the reference file

File: mixer/conf/application.conf [hunk 2]
```diff
@@ -46,3 +46,6 @@ numConfirmation = 3 // increasing this will cause slower mixes but more confiden
 maxOutputs = 10 // maximum number of outputs in one txs.
 maxInputs = 6 // maximum number of inputs in one txs.
 hopRounds = 0 // number of hops when withdrawing a box from mix
+
+bestBlockId = "8c53b6bafc56505d69a4dd53dd39731181de888047200dc464e9593d1bad832d"
```
Comment (anon2020s) [inline mixer/conf/application.conf]: Move to `mixer/conf/reference.conf` and use a valid blockId in mainnet

File: mixer/reference.conf [hunk 3]
```diff
@@ -59,6 +59,9 @@ fees = {
   halfTokenTx = 1287000
   fullTokenTx = 1666000
   ageusd = 5000000
+
+  stealthTransactionFeePercent = 0
```
Comment (anon2020s) [inline mixer/reference.conf:63]: Should be remove from reference.conf

File: mixer/test/controllers/ApiControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package controllers
```
Comment (anon2020s) [inline mixer/test/controllers/ApiControllerSpec.scala]: What is this file?

File: mixer/app/dao/stealth/ScanDAO.scala [hunk 5]
```diff
@@ -0,0 +1,103 @@
+package dao.stealth
+
```
Comment (anon2020s) [inline mixer/app/dao/stealth/ScanDAO.scala]: According to your [solution](https://github.com/ergoMixer/ergoMixBack/blob/e2febce5d99f0c64750ff803e7802a7744f26116/mixer/app/stealth/NodeProcess.scala#L76) for scanning boxes do not need to a table for register scans

File: mixer/app/services/ScannerTask.scala [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      step(syncHeight)
+    }
+  }
+
+
+
+  def scanRegister(): Unit = {
+    try {
+      if (scanDAO.selectAll.scans.isEmpty) {
+        val scan = readJsonFile("../conf/Stealth_newScan.json").stripMargin
```
Comment (anon2020s) [inline mixer/app/services/ScannerTask.scala]: don't need to save a scanRule should remove this scenario

File: mixer/app/services/StealthScheduledJobs.scala [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  private val logger: Logger = Logger(this.getClass)
+
+  /**
+   * periodically start scanner, task.
+   */
+  def receive: PartialFunction[Any, Unit] = {
+    case StealthJobsInfo.initBestBlockInDb =>
+      logger.info("Start job Store Best Block task.")
+      Try(initBestBlock.store_block())
+    case StealthJobsInfo.registerScan =>
```
Comment (anon2020s) [inline mixer/app/services/StealthScheduledJobs.scala]: remove register scan task

File: mixer/app/services/StealthStartupService.scala [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  val jobs: ActorRef = system.actorOf(Props(new StealthJobs(scannerTask, initBestBlockTask)), "scheduler")
+
+  override def onStart(): Unit ={
+    system.scheduler.scheduleOnce(
+      delay = 0.seconds,
+      receiver = jobs,
+      message = StealthJobsInfo.initBestBlockInDb
+    )
+
+    system.scheduler.scheduleOnce(
```
Comment (anon2020s) [inline mixer/app/services/StealthStartupService.scala]: remove register scan task

File: mixer/app/stealth/StealthContract.scala [hunk 9]
```diff
@@ -0,0 +1,363 @@
...
+
+            val txB = ctx.newTxBuilder()
+            val fee = feeCalculator(inputBoxList.toList)
+
+            val output = txB.outBoxBuilder()
+              .value(inputBoxList.map(_.getValue.toLong).sum - fee)
+              .contract(new ErgoTreeContract(WalletHelper.getAddress(address).script))
+              .build()
+
+            val tx = txB.boxesToSpend(inputBoxList.toList.asJava)
```
Comment (anon2020s) [inline mixer/app/stealth/StealthContract.scala:286]: If there are tokens in the boxes, they are not considered

File: mixer/test/stealth/StealthContractSpec.scala [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+    // reading ergo full block data
+    val extractionResult = readingErgoFullBlockData()
+    // storing outputs
+    stealthDaoContext.extractionResultDAO.storeOutputsAndRelatedData(extractionResult.createdOutputs, ExtractedBlock(header))
+    stealthDaoContext.extractionResultDAO.spendOutputsAndStoreRelatedData(extractionResult.extractedInputs)
+    stealthObj.updateBoxesStealthId(extractionResult.createdOutputs)
+
+    val addresses = stealthObj.getStealthAddresses
+
+    addresses.nonEmpty should be (true)
```
Comment (anon2020s) [inline mixer/test/stealth/StealthContractSpec.scala]: The expected purpose is different from the check performed

File: mixer/test/stealth/StealthContractSpec.scala [hunk 11]
```diff
@@ -0,0 +1,336 @@
+package stealth
```
Comment (anon2020s) [inline mixer/test/stealth/StealthContractSpec.scala:1]: Need more tests, also implemented unit tests need improvement

**Review by anon2020s — CHANGES_REQUESTED** [review]
Please check comments, also check CI runner log

File: mixer/app/Configs.scala (hunk 1 above)
Comment (aragogi) [reply mixer/app/Configs.scala]: Done!

File: mixer/conf/application.conf (hunk 2 above)
Comment (aragogi) [reply mixer/conf/application.conf]: Done!

File: mixer/reference.conf (hunk 3 above)
Comment (aragogi) [reply mixer/reference.conf:63]: Done!

File: mixer/test/controllers/ApiControllerSpec.scala (hunk 4 above)
Comment (aragogi) [reply mixer/test/controllers/ApiControllerSpec.scala]: Removed.

File: mixer/app/dao/stealth/ScanDAO.scala (hunk 5 above)
Comment (aragogi) [reply mixer/app/dao/stealth/ScanDAO.scala]: Removed.

File: mixer/app/services/ScannerTask.scala (hunk 6 above)
Comment (aragogi) [reply mixer/app/services/ScannerTask.scala]: Fixed!

File: mixer/app/services/StealthScheduledJobs.scala (hunk 7 above)
Comment (aragogi) [reply mixer/app/services/StealthScheduledJobs.scala]: removed.

File: mixer/app/services/StealthStartupService.scala (hunk 8 above)
Comment (aragogi) [reply mixer/app/services/StealthStartupService.scala]: removed.

File: mixer/app/stealth/StealthContract.scala (hunk 9 above)
Comment (aragogi) [reply mixer/app/stealth/StealthContract.scala:286]: Handled.

File: mixer/test/stealth/StealthContractSpec.scala (hunk 10 above)
Comment (aragogi) [reply mixer/test/stealth/StealthContractSpec.scala]: Fixed!

File: mixer/test/stealth/StealthContractSpec.scala (hunk 11 above)
Comment (aragogi) [reply mixer/test/stealth/StealthContractSpec.scala:1]: Handled!

**Comment by anon2020s** [comment]
https://github.com/ergoMixer/ergoMixBack/commit/f95d6a2406798ac494fc5517e052782a5478453b

**Pull request closed without merging**